src/main/scala/net/snowflake/spark/snowflake/SnowflakeWriter.scala

```diff
@@ -288,10 +288,52 @@ private[snowflake] class SnowflakeWriter(
   /**
     * Generate the COPY SQL command
     */
-  private def copySql(sqlContext: SQLContext,
+  private def copySql(data: DataFrame,
+                      saveMode: SaveMode,
                       params: MergedParameters,
                       filesToCopy: (String, String),
                       tempStage: Option[String]): String = {
+
+    if (saveMode != SaveMode.Append && params.columnMap.isDefined)
+      throw new UnsupportedOperationException("The column mapping only works in append mode.")
+
+    def getMappingToString(list: Option[List[(Int, String)]]): String = {
```

Comment:
Where are we checking compatibility with the target table? What if the columns don't exist in the target table?